pkgdb/src/pkgdb/write.cc

```diff
@@ -434,73 +435,118 @@ PkgDb::setPrefixDone( const flox::AttrPath & prefix, bool done )
+  if ( this->completedAttrSet( parentId ) ) { return true; }
 
   bool tryRecur = prefix.front() != "packages";
 
   debugLog( nix::fmt( "evaluating package set '%s'",
                       concatStringsSep( ".", prefix ) ) );
 
-  /* Scrape loop over attrs */
-  for ( nix::Symbol & aname : cursor->getAttrs() )
+  auto processAttrib
+    = [this, &syms, tryRecur]( const flox::Cursor &      childCursor,
+                               const flox::AttrPath &    prefix,
+                               const flox::pkgdb::row_id parentId,
+                               const nix::Symbol &       aname,
+                               Todos &                   todo ) -> bool
+  {
+    try
+      {
+        if ( childCursor->isDerivation() )
+          {
+            this->addPackage( parentId, syms[aname], childCursor );
+            return false;
+          }
+        else if ( ! tryRecur ) { return false; }
+        else if ( auto maybe
+                  = childCursor->maybeGetAttr( "recurseForDerivations" );
+                  ( ( maybe != nullptr ) && maybe->getBool() )
+                  /* XXX: We explicitly recurse into `legacyPackages.*.darwin'
+                   *      due to a bug in `nixpkgs' which doesn't set
+                   *      `recurseForDerivations' attribute correctly. */
+                  || ( ( prefix.front() == "legacyPackages" )
+                       && ( syms[aname] == "darwin" ) ) )
+          {
+            flox::AttrPath path = prefix;
+            path.emplace_back( syms[aname] );
+            row_id childId = this->addOrGetAttrSetId( syms[aname], parentId );
+            todo.emplace( std::make_tuple( std::move( path ),
+                                           std::move( childCursor ),
+                                           childId ) );
+            return true;
+          }
+        else { return false; }
+      }
+    catch ( const nix::EvalError & err )
+      {
+        /* Ignore errors in `legacyPackages' */
+        if ( tryRecur )
+          {
+            /* Only print eval errors in "debug" mode. */
+            nix::ignoreException( nix::lvlDebug );
+            return false;
+          }
+        else { throw; }
+      }
+  };
+
+  auto   allAttribs   = cursor->getAttrs();
+  size_t startIdx     = pageIdx * pageSize;
+  size_t thisPageSize = startIdx + pageSize < allAttribs.size()
+                          ? pageSize
+                          : allAttribs.size() % pageSize;
+  bool   lastPage     = thisPageSize < pageSize;
+  auto   page
+    = std::views::counted( allAttribs.begin() + startIdx, thisPageSize );
+
+  for ( nix::Symbol & aname : page )
     {
-      if ( syms[aname] == "recurseForDerivations" ) { continue; }
+      if ( auto lvl = nix::lvlTalkative; lvl <= nix::verbosity )
+        {
+          const std::string pathS
+            = concatStringsSep( ".", prefix ) + "." + syms[aname];
+          traceLog( nix::fmt( "Processing attribute path: %s.", pathS ) );
```

Comment:
```suggestion
          traceLog( nix::fmt( "Processing attribute path: '%s'.", pathS ) );
```
Since `pathS` is `.` delimited quotes help avoid ambiguity/confusion when printing the path.